gems/aws-sdk-s3/spec/client/region_detection_spec.rb

```diff
@@ -85,6 +85,82 @@ module S3
         end
 
       end
+
+      describe "Client Side Monitoring" do
+        let(:stub_publisher) do
+          StubPublisher.new
+        end
+
+        let(:client_opts) {{
+          region: 'us-west-2',
+          credentials: Credentials.new('akid', 'secret'),
+          client_side_monitoring: true,
+          client_side_monitoring_publisher: stub_publisher
+        }}
+
+        let(:client) { Client.new(client_opts) }
+
+        before(:each) do
+          stub_publisher.metrics = []
+        end
+
+        it 'updates fqdn and region in metrics when an HTTP 400 redirect occurs' do
+          stub_request(:put, 'https://bucket.s3.us-west-2.amazonaws.com/key').
+            to_return(status: [400, 'Bad Request'], body: "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AuthorizationHeaderMalformed</Code><Message>The authorization header is malformed; the region 'us-east-1' is wrong; expecting 'eu-central-1'</Message><Region>eu-central-1</Region><RequestId>531B68B3613F5C96</RequestId><HostId>TMnOREh0Ms0touCRX0XkJinw7xqsF0v/iFyA+nCC4d3PpF+k2oekWlSrUk+8d2/rvcnEv2QXer0=</HostId></Error>")
```

Comment:
This doesn't affect test result, but just a typo:

the region 'us-east-1' is wrong
should be 'us-west-2'?

Reply:
Ahh yes, looks like i only fixed this in one place. Updating.